keras_core/backend/torch/random.py

```diff
@@ -126,7 +126,10 @@ def _get_concrete_noise_shape(inputs, noise_shape):
 
 
 def dropout(inputs, rate, noise_shape=None, seed=None):
-    if noise_shape is not None:
+    if noise_shape is not None or not (
+        (seed is None)
+        or (isinstance(seed, SeedGenerator) and seed.initialized_with_none)
+    ):
```

Comment:
You can do plain chaining of the 3 `or`s

Reply:
`A or not (B or C)` is equivalent to `A or (not B and not C)`.
So I just chain them with `and` and `or`.
Seems not possible to chain them with all `or`s.